src/all_optimizations.py

```diff
@@ -848,33 +851,74 @@ def dlc(DATA_DIR, dlc_thresh):
 
 if __name__ == '__main__':
     parser = ArgumentParser(description='All Optimizations')
-    parser.add_argument('--data_dir', type=str, help='The file path to the flick/run to be optimized')
-    parser.add_argument('--start_frame', type=int, default=1, help='The frame at which the optimized reconstruction will start')
-    parser.add_argument('--end_frame', type=int, default=-1, help='The frame at which the optimized reconstruction will end')
-    parser.add_argument('--dlc_thresh', type=float, default=0.8, help='The likelihood of the dlc points below which will be excluded from the optimization')
-    parser.add_argument('--plot', action='store_true', help='Show the plots')
+    parser.add_argument('--data_dir', type=str, help='The file path to the flick/run to be optimized.')
+    parser.add_argument('--start_frame', type=int, default=1, help='The frame at which the optimized reconstruction will start.')
+    parser.add_argument('--end_frame', type=int, default=-1, help='The frame at which the optimized reconstruction will end. If it is -1, start_frame and end_frame are automatically set.')
```

Comment:
It might be better if we use the `start_frame` as the trigger for the automatic frame selection. That way we can use `end_frame=-1` as an indication to use the entire video. So we have the options to:

- Manually set `start_frame` and `end_frame` to values ` > 0`.
- Use the entire video segment by setting `end_frame = -1`.
- Automatically select the "best" frames based on the first time (and last time) the markers are all visible by setting `start_frame = -1`.

Just a thought.